controllers/nstemplateset/nstemplateset_controller_test.go

```diff
@@ -1372,74 +1372,31 @@ func TestDeleteNSTemplateSet(t *testing.T) {
 		require.Empty(t, result)
 	})
 
-	t.Run("delete when there is no finalizer", func(t *testing.T) {
```

Comment:
Deleted this testcase because it was no longer possible to test this scenario which required to observe the object in a state when it has a deletion timestamp but no finalizer, because the object would get deleted before it can be observed.
If we still need this test case - we can try looking at [Interceptors](https://github.com/kubernetes-sigs/controller-runtime/pull/2306) in the following PR.

Reply:
Yes, precisely.